src/Storages/Kafka/KafkaSource.cpp

```diff
@@ -224,6 +243,26 @@ Chunk KafkaSource::generateImpl()
                     }
                 }
             }
+            if (is_dead_letter && exception_message)
```

Comment:
Why is the `exception_message` part is important? I think when `dead_letter` is `true`, then `exception_message` is always set. Even if not, then we should still send the message to dead letter.
```suggestion
            if (is_dead_letter)
```